src/components/transportation-modes/TransportModes.tsx

```diff
@@ -40,28 +42,42 @@ export const getTransportModeText = (
   modes: TransportModePair[],
   t: TranslateFunction,
   modesDisplayLimit: number = 2,
+  unknownModeText?: string,
 ): string => {
   const modesCount: number = modes.length;
 
   if (!modes) return '';
   if (modesCount > modesDisplayLimit) {
     return t(FareContractTexts.transportModes.multipleTravelModes);
   }
-  return modes
-    .map((tm) => t(FareContractTexts.transportMode(tm.mode, tm.subMode)))
-    .filter(removeDuplicateStringsFilter)
-    .join('/');
+  if (unknownModeText && modes.map((m) => m.mode).includes('unknown')) {
+    return unknownModeText;
+  }
```

Comment:
Is this handling of `unknownModeText` intuitive? I kinda feel if we somehow get `rail` and `unknown` as transport modes, we would kinda like to still show the train icon and train text? I know it makes sense in the context it is used now, but maybe not if we think of `TransportModes` as a shared generic component?